File: app/src/main/java/me/ykrank/s1next/view/adapter/delegate/HomeThreadAdapterDelegate.java
package me.ykrank.s1next.view.adapter.delegate;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import me.ykrank.s1next.data.api.model.HomeThread;
import me.ykrank.s1next.databinding.ItemHomeThreadBinding;
import me.ykrank.s1next.viewmodel.HomeThreadViewModel;

/**
 * Created by ykrank on 2017/2/4.
 */

public class HomeThreadAdapterDelegate extends BaseAdapterDelegate<HomeThread, HomeThreadAdapterDelegate.BindingViewHolder> {

    public HomeThreadAdapterDelegate(Context context) {
        super(context);
    }

    @NonNull
    @Override
    protected Class<HomeThread> getTClass() {
        return HomeThread.class;
    }

    @Override
    public void onBindViewHolderData(HomeThread thread, int position, @NonNull BindingViewHolder holder) {
        ItemHomeThreadBinding binding = holder.binding;
        binding.getModel().thread.set(thread);
        binding.executePendingBindings();
    }

    @NonNull
    @Override
    protected RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent) {
        ItemHomeThreadBinding binding = ItemHomeThreadBinding.inflate(mLayoutInflater, parent, false);
        binding.setModel(new HomeThreadViewModel());
        return new HomeThreadAdapterDelegate.BindingViewHolder(binding);
    }

    static final class BindingViewHolder extends RecyclerView.ViewHolder {

        private final ItemHomeThreadBinding binding;

        public BindingViewHolder(ItemHomeThreadBinding binding) {
            super(binding.getRoot());

            this.binding = binding;
        }
    }
}

File: app/src/main/java/me/ykrank/s1next/view/adapter/delegate/HomeThreadAdapterDelegate.kt
package me.ykrank.s1next.view.adapter.delegate

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import com.github.ykrank.androidtools.ui.adapter.simple.SimpleRecycleViewHolder

import me.ykrank.s1next.data.api.model.HomeThread
import me.ykrank.s1next.databinding.ItemHomeThreadBinding
import me.ykrank.s1next.viewmodel.HomeThreadViewModel

/**
 * Created by ykrank on 2017/2/4.
 */

class HomeThreadAdapterDelegate(context: Context) : BaseAdapterDelegate<HomeThread, SimpleRecycleViewHolder<ItemHomeThreadBinding>>(context, HomeThread::class.java) {

    override fun onBindViewHolderData(thread: HomeThread, position: Int, holder: SimpleRecycleViewHolder<ItemHomeThreadBinding>, payloads: List<Any>) {
        val binding = holder.binding
        binding.model?.thread?.set(thread)
    }

    override fun onCreateViewHolder(parent: ViewGroup): androidx.recyclerview.widget.RecyclerView.ViewHolder {
        val binding = ItemHomeThreadBinding.inflate(mLayoutInflater, parent, false)
        binding.model = HomeThreadViewModel()
        return SimpleRecycleViewHolder<ItemHomeThreadBinding>(binding)
    }
}
